### ISSUE TITLE
Bug: Ingredient with `Only check if any amount is in stock` set and unit conversion: Wrong amount on shopping list

### ISSUE THREAD

**Issue by heinrich-ulbricht** [grocy/grocy#1718]
EDIT by @berrnd:
When the ingredient option `Only check if any amount is in stock` is checked, the amount on the shopping list is wrong when unit conversions are involved (since the ingredient amount is treated as `1` internally for everything).

----

**Overall goal**

Track a product in _gram_ but add it as _piece_ to recipes. E.g. buy a kilo of tomatoes, but have a recipe say "5 tomatoes". Then when adding the ingredients to the shopping list the correct amount should be added (either gram or pieces).

**Setup for this case**

* add "Tomatogram" as new product
  * set its units to _gram_ (note: _not_ piece)
  * add QU conversions so that one piece is 10 gram and 1 gram is 0.1 piece
* create recipe for 1 serving
  * add sole ingredient "Tomatogram", change its unit to _piece_ and set amount to 1 (so 1 tomato is needed)
* add this recipe to shopping list

**What do I observe in the shopping list?**

1 gram of "Tomatogram" is being added to the shopping list. This is wrong.

_Note: I tested on the prerelease demo._

**What do I expect in the shopping list?**

I expect to see 10 gram of "Tomatogram" because the recipe contains "1 piece" and this - per the QU conversion - corresponds to "10 gram".

_Note: This might not be a bug but a feature request (that is, respecting the QU conversion). But it feels like an error that 1 **piece** of Tomatogram is added as 1 **gram** to the shopping list. The issue title might have to be changed depending on how you interpret this issue._

**Comment by heinrich-ulbricht** [comment]
Ah thanks for the hint, I think I never (tried to) understood what those private instances are about.

Here's the sample, for me this reproduces the error:

https://test-9k5pakpg6a8ejii3839vbh.demo-prerelease.grocy.info

**Comment by heinrich-ulbricht** [comment]
I see, the difference is the "Only check if any amount is in stock" which I checked on my instance.

In another instance I verified that this makes the difference.

**Comment by berrnd** [comment]
`Only check if any amount is in stock` was originally supposed for things like spices where it's not suitable to buy the amount per recipe when needed. Internally for stock fulfillment checking and the rest, just `> 0` is used instead of the entered amount in that case (as the help text for that option kind of implies).

So ok, let's track it as a bug that regardless of that option, the amount on the shopping list should be correct (however if that makes practically sense or not, for me not so much...).

**Comment by heinrich-ulbricht** [comment]
Just as side note how I discovered the bug: Before learning about the QU conversion option I used the "Only check if any amount is in stock" option to make all units available for the ingredient. In my particular case I only saw "gram" but wanted to use "piece" for tomatoes. So I checked the box and suddenly I could choose piece - cool. (This was before configuring the QU conversion for tomatoes.) Later I discovered the QU conversion option, configured it and so from that point on I wouldn't have needed the "Only check if any amount is in stock" option anymore. But it was still checked and made me discover the bug.

So yeah. A rather low-priority bug that is rather confusing than really blocking anything. Avoidable if one knows how and where to put the right data in.
